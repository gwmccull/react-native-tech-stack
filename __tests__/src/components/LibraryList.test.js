jest.mock('../../../src/components/ListItem', () => 'ListItem');

import React from 'react';
import renderer from 'react-test-renderer';

import { LibraryList, mapStateToProps } from '../../../src/components/LibraryList';

const defaultProps = {
  libraries: [],
};

describe('LibraryList', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <LibraryList {...defaultProps} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
})

describe('LibraryList functions', () => {
  it('renderRow', () => {
    const library = {};
    const wrapper = renderer.create(
      <LibraryList {...defaultProps} />
    ).getInstance();
    expect(wrapper.renderRow(library)).toMatchSnapshot();
  });

  it('mapStateToProps', () => {
    const libraries = [];
    const state = {
      libraries,
    };
    const result = mapStateToProps(state);
    expect(result).toEqual({ libraries });
  })
})
