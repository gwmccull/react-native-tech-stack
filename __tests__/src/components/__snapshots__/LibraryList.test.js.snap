// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LibraryList functions renderRow 1`] = `
<ListItem
  library={Object {}}
/>
`;

exports[`LibraryList renders correctly 1`] = `
<RCTScrollView
  dataSource={
    ListViewDataSource {
      "items": 0,
    }
  }
  renderRow={[Function]}
  renderScrollComponent={[Function]}
>
  <View />
</RCTScrollView>
`;
